compiler/rustc_trait_selection/src/solve/assembly/structural_traits.rs

```diff
@@ -131,6 +131,10 @@ pub(in crate::solve) fn instantiate_constituent_tys_for_sized_trait<'tcx>(
         | ty::Dynamic(_, _, ty::DynStar)
         | ty::Error(_) => Ok(vec![]),
 
+        ty::Alias(ty::Weak, alias) => {
```

Comment:
I don't think we need to do this (or any of the other structural impls below) -- if we support normalizing weak aliases via a projection predicate (which we should), then when we have `Weak: Sized`, then we'll assemble the correct candidate via [`assemble_candidates_after_normalizing_self_ty`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_trait_selection/solve/eval_ctxt/struct.EvalCtxt.html#method.assemble_candidates_after_normalizing_self_ty).